pkg/controller/directvolumemigration/rsync.go

```diff
@@ -565,6 +565,36 @@ func (t *Task) getRsyncRoute(namespace string) (string, error) {
 	return route.Spec.Host, nil
 }
 
+//Returns a map of PVCNamespacedName to the pod.NodeName
+func (t *Task) getPVCNodeNameMap() (map[string]string, error) {
+	nodeNameMap := map[string]string{}
+	pvcMap := t.getPVCNamespaceMap()
+	nsPodList := corev1.PodList{}
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return nil, err
+	}
+
+	for ns, _ := range pvcMap {
+		err = srcClient.List(context.TODO(), k8sclient.InNamespace(ns), &nsPodList)
+		if err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
Won't this just end up with the pods for the final namespace included? I think the pod list loop should be inside the above loop.

Reply:
Thanks for pointing it out. I had earlier done the same but was trying something else and missed out the multiple ns scenario. Updated the PR.